### ISSUE TITLE
[Enhancement] Refactoring user settings parsing

### ISSUE THREAD

**Issue by koszeggy** [NickeManarin/ScreenToGif#1093]
Parsing user settings is a pain now as whenever a new enum or nullable type is used it must be handled manually.

A need for refactoring is also [indicated](https://github.com/NickeManarin/ScreenToGif/blob/84a0a739163ab61bc2ff7ea7669fdc3549e5ca38/ScreenToGif.Util/Settings/UserSettings.cs#L194) in the source code:

    //Do you know any other way to achieve this? Contact me via Github.

As I [mentioned](https://github.com/NickeManarin/ScreenToGif/issues/1069#issuecomment-1034874240) I already addressed the issue during the development of #1069 because it affected my changes as well. But since the solution was dependent on #1069 (because I created a sub-branch from its feature branch and because it uses one of the libraries that was added for #1069) I waited with a new pull request until #1069 was merged.

Git was a bit confused though when I tried to merge the [refactoring branch](https://github.com/koszeggy/ScreenToGif/tree/UserSettingsRefactoring) to `dev` because all of the changes of #1069 appeared again for review (even though these changes are merged now into `dev`), so in the end I canceled the pull request. I can try it again during the next week by creating a new branch directly from `dev` and reintroducing the changes (only `UserSettings.cs` is affected) so it will be clean enough for git, too.

Until then, feel free to assign the issue to me.

**Comment by koszeggy** [comment]
I created a [pull request](https://github.com/NickeManarin/ScreenToGif/pull/1094)

Please test it thoroughly. I tested it just by trying to generate the largest possible user settings (I opened as many menus I could).

The only part that still might need manual adjustment is for the XAML element nodes that refer to built-in WPF types. Now types are resolved from the `System.Windows`, `System.Windows.Media` and `System.Windows.Ink` namespaces, which still should be [specified explicitly](https://github.com/koszeggy/ScreenToGif/blob/aca1c33b1d58e9a56ec0f562ac27f88094c11cbe/ScreenToGif.Util/Settings/UserSettings.cs#L234) if needed.

**Comment by NickeManarin** [comment]
Thank you!
